### PR TITLE
Expose Top level API through zombienet-sdk crate

### PR DESCRIPTION
**Pull request by pgherveou**
Following up on take 1 https://github.com/paritytech/zombienet-sdk/pull/124

Apart from better logs, I realize that one of the only thing missing now for us to start consuming this project  is a top level package available (with it's dependencies) on crates.io. This PR serves mainly as a discussion point to get there.
It would be great if we could publish an early version (0.0.1?) on crates.io so that we can start using this asap.

### CODE REVIEW

File: crates/sdk/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+use async_trait::async_trait;
+pub use configuration::{NetworkConfig, NetworkConfigBuilder, RegistrationStrategy};
+pub use orchestrator::{errors::OrchestratorError, network::Network, AddNodeOpts, Orchestrator};
+use provider::NativeProvider;
+use support::{fs::local::LocalFileSystem, process::os::OsProcessManager};
+
+#[async_trait]
+pub trait Spawner {
```
Comment (pepoviola) [inline crates/sdk/src/lib.rs:8]: I think we can rethink this `naming`, something like `NetworkConfigSpawnExt` or just `NetworkConfigExt`. 
@wirednkod / @l0r1s / @pgherveou wdyt?

**Comment by l0r1s** [comment]
Hi @pgherveou, sure ! (and some docs probably)

File: crates/sdk/src/lib.rs (hunk 1 above)
Comment (l0r1s) [reply crates/sdk/src/lib.rs:8]: Yes, we can do this temporarily, not a issue to change this in the future :+1:

Comment (pgherveou) [reply crates/sdk/src/lib.rs:8]: 👍  NetworkConfigExt sounds a bit better

**Comment by pgherveou** [comment]
@pepoviola most urgent missing bit, apart from this top level package is fixing this todo! that prevent us from defining a custom command.

https://github.com/paritytech/zombienet-sdk/blob/m/crates/orchestrator/src/generators/chain_spec.rs#L364

The other thing is: `get_node` only return validator node for now, should we adapt the code like this so that we can get reference to any node, including parachain collators?

`add_para` probably need to be adjusted like this 
```diff
     pub(crate) fn add_para(&mut self, para: Parachain) {
+        para.collators.iter().for_each(|collator| {
+            self.nodes_by_name
+                .insert(collator.name.clone(), collator.clone());
+        });
         self.parachains.insert(para.para_id, para);
     }
```

**Comment by pepoviola** [comment]
> @pepoviola most urgent missing bit, apart from this top level package is fixing this todo! that prevent us from defining a custom command.
> 
> https://github.com/paritytech/zombienet-sdk/blob/m/crates/orchestrator/src/generators/chain_spec.rs#L364
> 
> The other thing is: `get_node` only return validator node for now, should we adapt the code like this so that we can get reference to any node, including parachain collators?
> 
> `add_para` probably need to be adjusted like this
> 
> ```diff
>      pub(crate) fn add_para(&mut self, para: Parachain) {
> +        para.collators.iter().for_each(|collator| {
> +            self.nodes_by_name
> +                .insert(collator.name.clone(), collator.clone());
> +        });
>          self.parachains.insert(para.para_id, para);
>      }
> ```

Thanks @pgherveou, I just open this https://github.com/paritytech/zombienet-sdk/issues/127 to make all the node available through `get_node`. Related to the chain_spec, did you need to use a custom command to generate it?

Thx!

**Review by pepoviola — APPROVED** [review]

**Pull request merged**